### PR TITLE
PCH Excerpt Suggestions: Allow editing when reviewing

### PR DESCRIPTION
**Pull request by acicovic**
## Description
This PR introduces code to allow editing the excerpt in the PCH Excerpt Suggestions tool while in review mode.

As part of this PR, we have grouped some excerpt state variables in an `excerptData` object. We're also handling updates to the textbox that come from keyboard shortcuts and aren't caught by the `onChange` event.

## Motivation and context
Make Excerpt Suggestions usage more intuitive.

## How has this been tested?
Manually tested.

## Summary by CodeRabbit

- **New Features**
  - Enhanced UI interaction for word count and excerpt changes.

- **Refactor**
  - Improved state management for excerpt data.
  - Refactored word count calculation and display logic.

- **Style**
  - Removed `textarea:read-only` styling from the excerpt textarea.

### CODE REVIEW

File: src/content-helper/excerpt-generator/components/excerpt-panel.tsx
Comment (acicovic) [reply src/content-helper/excerpt-generator/components/excerpt-panel.tsx]: Isn't this handled by setOnChangeFired?

Comment (acicovic) [reply src/content-helper/excerpt-generator/components/excerpt-panel.tsx]: @coderabbitai resolve

File: src/content-helper/excerpt-generator/components/excerpt-panel.tsx
Comment (vaurdan) [reply src/content-helper/excerpt-generator/components/excerpt-panel.tsx:106]: Although this is a pretty valid point, I don't think it would be an issue in this case, since it's restricted to the excerpt textarea, and adding a debounce would prevent the word count to update in real-time - unless the debounce interval would be set to very low, but that kinda defeats the purpose. 

I think we can safely ignore this suggestion. What do you think @acicovic?

Comment (acicovic) [reply src/content-helper/excerpt-generator/components/excerpt-panel.tsx:106]: Can you explain what this code will do?

File: src/content-helper/excerpt-generator/components/excerpt-panel.tsx
Comment (vaurdan) [reply src/content-helper/excerpt-generator/components/excerpt-panel.tsx]: Unless I'm missing something don't see how this is any different from the current approach.

File: src/content-helper/excerpt-generator/components/excerpt-panel.tsx
Comment (vaurdan) [inline src/content-helper/excerpt-generator/components/excerpt-panel.tsx:196]: If we change the generated excerpt and not approve it, nor discard it, will it not still be saved to the post? I think the `editPost` function should only be called when `excerptData.newExcerptGenerated` is false/empty.

File: src/content-helper/excerpt-generator/components/excerpt-panel.tsx
Comment (acicovic) [reply src/content-helper/excerpt-generator/components/excerpt-panel.tsx:106]: Oh, we posted at the same time 🙂

That was what I was pondering about. This might make a slight difference in some cases, but I think it's marginal, unless we set that to 1 second or something. Let's ignore this suggestion.

@coderabbitai resolve

**Review by vaurdan — COMMENTED** [review]
Thanks for working on this one! Looks great, I just have added a single comment, and once it is addressed I can go ahead and approve it.

File: src/content-helper/excerpt-generator/components/excerpt-panel.tsx
Comment (acicovic) [reply src/content-helper/excerpt-generator/components/excerpt-panel.tsx:196]: Do you think that this saving would take place due to `editPost()`? In my testing, the only way to persist the excerpt is to press the post's `Update` button. `editPost()` doesn't seem to have a permanent effect.

Comment (vaurdan) [reply src/content-helper/excerpt-generator/components/excerpt-panel.tsx:196]: Ah yeah, that sounds right! This would only happen if the user wouldn't reject nor accept the excerpt, but would still save the post. So maybe not really an issue, I will leave that to your discretion 🙁 

Edit: Although, not sure what would happen when the post autosaves.

**Review by vaurdan — APPROVED** [review]
Approved per my last reply.

File: src/content-helper/excerpt-generator/components/excerpt-panel.tsx
Comment (acicovic) [reply src/content-helper/excerpt-generator/components/excerpt-panel.tsx:196]: > Edit: Although, not sure what would happen when the post autosaves.

Interesting observation. I'll look into it.

Comment (acicovic) [reply src/content-helper/excerpt-generator/components/excerpt-panel.tsx:196]: Testing with a draft post, I can confirm that the textarea's temporary value (shown but not dismissed or accepted) isn't persisted when auto-saving takes place.

Given this, I think we're safe.

Comment (acicovic) [reply src/content-helper/excerpt-generator/components/excerpt-panel.tsx:196]: There's an exception. If the user has typed something manually in that state, changes persist.

Comment (acicovic) [reply src/content-helper/excerpt-generator/components/excerpt-panel.tsx:196]: This seems to be standard WP behavior though. When we type something manually, it gets persisted.

Comment (acicovic) [reply src/content-helper/excerpt-generator/components/excerpt-panel.tsx:196]: So there's a potential for unexpected behavior. I'll leave this open for the time being.

Comment (acicovic) [reply src/content-helper/excerpt-generator/components/excerpt-panel.tsx:196]: This should be fixed now.

**Comment by acicovic** [comment]
@coderabbitai resolve

**Pull request merged**
